### DISCUSSION TITLE
Auto-generate data docs - Convert table schema, data resource and data package to markdown

### DISCUSSION THREAD

**Discussion by rufuspollock** [frictionlessdata/frictionlessdata.io#665] [Ideas]
When I get a table schema (or data resource or data package) I want to convert it to markdown for presentation or previewing e.g. in README or converting to html so that i can create human readable documentation of my data quickly (and repeatedly)

e.g. I want to create README for data package and i want to insert table schema summary in the README.

## Existing work

* See `data info` command on data-cli https://github.com/datopian/data-cli (js based)
* https://framagit.org/opendataschema/table-schema-to-markdown/ (js based)

**Comment by johanricher** [comment]
Hi! We actually built a library to do that for Table Schemas, still very early version obviously but used in production ([see this Gitbook for example](https://scdl.opendatafrance.net/)):
https://framagit.org/opendataschema/table-schema-to-markdown/

Thera's also a Python library but it's deprecated now I think) https://github.com/AntoineAugusti/table-schema-to-markdown

**Reply by johanricher** [reply]
The Python library has been forked and updated here by @geoffreyaldebert https://github.com/geoffreyaldebert/table-schema-to-markdown, it's used in production by https://github.com/etalab/schema.data.gouv.fr/

**Reply by lwinfree** [reply]
That's awesome! Thanks for the update :-)

**Comment by lwinfree** [comment]
There's also this code that Carles wrote as part of the Schema-Collaboration tool fund: https://github.com/frictionlessdata/schema-collaboration/blob/master/SchemaCollaboration/datapackage_to_documentation/main.py (docs: https://github.com/frictionlessdata/schema-collaboration/tree/master/documentation)

**Comment by roll** [comment]
I think we need to provide the functions below to Frictionless-py:

```
schema.to_markdown
resource.to_markdown
package.to_markdown
```

Please let me know if someone is interested to PR based on the existent tools :+1:

**Reply by johanricher** [reply]
@geoffreyaldebert What do you think about integrating [`table-schema-to-markdown`](https://github.com/geoffreyaldebert/table-schema-to-markdown) into `frictionless-py`?

**Reply by geoffreyaldebert** [reply]
@johanricher @roll I think it would be cool yes and more generally practical to have all features around tableschema in the same package. So no problem for me.

**Reply by roll** [reply]
@geoffreyaldebert
Are you interested in working on the integration by yourself or it would work better for you if we handle it?

**Reply by geoffreyaldebert** [reply]
I think it would be better if you handle it if you are hurry. I'm not very available at the moment, I could work on that by the end of may but difficult for me before

**Reply by roll** [reply]
Cool, I've created a feature request - https://github.com/frictionlessdata/frictionless-py/issues/837. Let's see if we can start earlier or let us know when you'd like to take it over if we haven't started
